### PR TITLE
Changed all code and comments that used the phrase "sparse compiler" to instead use "sparsifier"

### PR DESCRIPTION
**Pull request by TimAtGoogle**
The changes in this p.r. mostly center around the tests that use the flag sparse_compiler (also: sparse-compiler).

### CODE REVIEW

File: mlir/test/Integration/Dialect/SparseTensor/python/tools/sparsifier.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
```
Comment (aartbik) [inline mlir/test/Integration/Dialect/SparseTensor/python/tools/sparsifier.py:1]: I would expect a rename from  "sparse_compiler.py"
(or at least a removal of that file then?)

Comment (TimAtGoogle) [reply mlir/test/Integration/Dialect/SparseTensor/python/tools/sparsifier.py:1]: I just pushed it with the file removed, so give it a second to check formattting.

Comment (aartbik) [reply mlir/test/Integration/Dialect/SparseTensor/python/tools/sparsifier.py:1]: I don't see the rename/removal yet?

File: mlir/test/Integration/Dialect/SparseTensor/python/tools/sparsifier.py [hunk 2]
```diff
@@ -2,19 +2,19 @@
...
 from mlir import execution_engine
 from mlir import ir
 from mlir import passmanager
 from typing import Sequence
 
 
-class SparseCompiler:
-    """Sparse compiler class for compiling and building MLIR modules."""
+class Sparsifier:
+    """sparsifier class for compiling and building MLIR modules."""
```
Comment (aartbik) [inline mlir/test/Integration/Dialect/SparseTensor/python/tools/sparsifier.py:14]: I would make this a capital S, as in "Sparsifier class for ..." to be consistent with the original

**Review by aartbik — APPROVED** [review]

**Pull request merged**